aten/src/THCUNN/generic/MultiMarginCriterion.cu

```diff
@@ -59,78 +59,81 @@ void THNN_(MultiMarginCriterion_updateOutput)(
   else if (input->dim() == 2)
   {
     int nframe = input->size(0);
-    THArgCheck(!target->is_empty() && (THTensor_nDimensionLegacyNoScalars(target) == 1) && (THTensor_sizeLegacyNoScalars(target, 0) == nframe), 3,
+    // allow zero-dim target for 2D input.
+    THArgCheck((input->size(1) != 0) && (THTensor_nDimensionLegacyNoScalars(target) == 1) && (THTensor_sizeLegacyNoScalars(target, 0) == nframe), 3,
                "inconsistent target size");
-    dim3 blocks(input->size(0));
-    dim3 threads(MULTIMARGIN_THREADS);
+    if (input->numel() != 0) {
```

Comment:
Again: can this check be inverted and a smaller short-circuit added for this case?

Reply:
Done.